rpcserver.go

```diff
@@ -66,17 +96,24 @@ func (s *server) AddPeer(
 		return makePlainResponse(false, fmt.Sprintf("Failed to generate peer key/ID with seed: %v", req.Seed)), nil
 	}
 
-	var replyMsg string
-	err = s.node.AddPeer(spanctx, mAddr)
-	success := (err == nil)
-	if success {
-		replyMsg = fmt.Sprintf("Added Peer %v:%v, pid=%v!", req.Ip, req.Port, targetPID)
-		// Tag the span with peer info
-		logger.SetTag(spanctx, "Added peer", fmt.Sprintf("%v:%v", req.Ip, req.Port))
-	} else {
-		replyMsg = fmt.Sprintf("Failed to add Peer %v:%v, pid=%v!", req.Ip, req.Port, targetPID)
+	peerid, targetAddr, err := parseAddr(mAddr)
+	if err != nil {
+		logger.FinishWithErr(spanctx, fmt.Errorf("Failed to parse peer address: %s, err: %v", mAddr, err))
+		log.Printf("Failed to parse peer address: %s, err: %v", mAddr, err)
+		return makePlainResponse(false, fmt.Sprintf("Failed to parse peer address: %s, err: %v", mAddr, err)), nil
 	}
-	return makePlainResponse(success, replyMsg), nil
+	s.node.Peerstore().AddAddr(peerid, targetAddr, pstore.PermanentAddrTTL)
+
+	if err := s.node.Connect(ctx, s.node.Peerstore().PeerInfo(peerid)); err != nil {
+		logger.FinishWithErr(spanctx, fmt.Errorf("Failed to connect to peer %v, err: %v", peerid, err))
+		log.Printf("Failed to connect to peer %v, err: %v", peerid, err)
```

Comment:
Might also change this to `logger.Errorf`? or it will be handled in that PR?

Reply:
Yes, it will be handled in #67 .